mvpa2/algorithms/group_clusterthr.py

```diff
@@ -236,205 +313,320 @@ def _train(self, ds):
-                bds = Dataset(clustermap, a=dsa)
-                # this function reverse-maps every sample one-by-one, hence no need
-                # to collect chunks of bootstrapped maps
-                cluster_sizes = get_cluster_sizes(bds, cluster_sizes)
-        else:
-            # Parallel execution
-            # same code as above, just restructured for joblib's Parallel
-            for jobres in Parallel(n_jobs=self.params.n_proc,
-                                   pre_dispatch=self.params.n_proc,
-                                   verbose=verbose_level_parallel)(
-                                       delayed(get_cluster_sizes)
-                                  (Dataset(np.mean(ds_samples[sidx],
-                                           axis=0)[None] > thrmap,
-                                           a=dsa))
-                                       for sidx in bcombos):
-                # aggregate
-                cluster_sizes += jobres
-        # store cluster size histogram for later p-value evaluation
-        # use a sparse matrix for easy consumption (max dim is the number of
-        # features, i.e. biggest possible cluster)
-        scl = dok_matrix((1, ds.nfeatures + 1), dtype=int)
-        for s in cluster_sizes:
-            scl[0, s] = cluster_sizes[s]
-        self._null_cluster_sizes = scl
+        # TODO: yoh: note that the same bcombos which were used to estimate thrmap
+        #       will now be used to estimate cluster sizes/metric.
+        #       Not sure if such double treatment of the same data doesn't provide
+        #       any bias.  We could  a) split chunks pool into two  b) create new
+        #       pool of bcombos for metric estimation
+        self._null_cluster_metric_counts = \
+            self._get_null_cluster_metric_counts(ds, thrmap, bcombos)
+
 
     def _call(self, ds):
         if len(ds) > 1:
+            # TODO: shouldn't we always demand all original data to enter?
             # average all samples into one, assuming we got something like one
             # sample per subject as input
-            avgr = mean_sample()
-            ds = avgr(ds)
+            # TODO:  our "func" invocation
+            measure = self.params.measure or mean_sample()
+            ds = measure(ds)
+
+        labeler = self._labeler
+        assert(len(ds) == 1)
+        ds_sample = ds.samples[0]
+        thrmap = self._thrmap
+
         # threshold input; at this point we only have one sample left
-        thrd = ds.samples[0] > self._thrmap
-        # mapper default
-        mapper = IdentityMapper()
-        # overwrite if possible
-        if hasattr(ds, 'a') and 'mapper' in ds.a:
-            mapper = ds.a.mapper
-        # reverse-map input
-        othrd = _verified_reverse1(mapper, thrd)
-        # TODO: what is your purpose in life osamp? ;-)
-        osamp = _verified_reverse1(mapper, ds.samples[0])
+        thrd = ds_sample > thrmap
```

Comment:
so far I have just left logic on that identical to before, but indeed we could as well provide an option to instruct which tail to deal with...
interested to help with this PR? got distracted to other errands, never finished this PR, but it should be done